ui/raidboss/popup-text.ts

```diff
@@ -818,15 +1040,19 @@ export class PopupText {
     }
   }
 
-  _onTriggerInternalPreRun(triggerHelper) {
-    if ('preRun' in triggerHelper.trigger)
-      triggerHelper.trigger.preRun(this.data, triggerHelper.matches, triggerHelper.trigger.output);
+  _onTriggerInternalPreRun(triggerHelper: TriggerHelper): void {
+    if ('preRun' in triggerHelper.trigger && triggerHelper.trigger.preRun) {
+      triggerHelper.trigger.preRun(
+          this.data,
+          triggerHelper.matches,
+          triggerHelper.trigger.output as Output);
+    }
   }
 
-  _onTriggerInternalDelaySeconds(triggerHelper) {
+  _onTriggerInternalDelaySeconds(triggerHelper: TriggerHelper): Promise<void> | undefined {
     const delay = 'delaySeconds' in triggerHelper.trigger ? triggerHelper.valueOrFunction(triggerHelper.trigger.delaySeconds) : 0;
-    if (!delay || delay <= 0)
-      return null;
+    if (!delay || delay <= 0 || typeof delay !== 'number')
```

Comment:
The issue with strings for delaySeconds is only on the trigger file side, because those cases all subtract a number from a string, thus returning a number.

This typeof check exists because the return type of valueOrFunction can be string as well, even though in this case it won't ever be a string. Without it, when trying to set the timeout, this error happens:

![image](https://user-images.githubusercontent.com/6119598/121702993-5c79db00-caa0-11eb-97cf-7739e3157a4c.png)

Reply:
The field definition in trigger.d.ts is for `TriggerField<Data, number>` which means that it should only return `number`.

This is visible when editing a typescript trigger file:

![image](https://user-images.githubusercontent.com/6119598/121725280-0bc1ac80-cab7-11eb-9eb7-1c3fd0fd162e.png)
![image](https://user-images.githubusercontent.com/6119598/121725312-1b40f580-cab7-11eb-91c4-e42ffdbe75b8.png)
![image](https://user-images.githubusercontent.com/6119598/121725340-24ca5d80-cab7-11eb-82ff-8e23e9f88364.png)

So either the definition in trigger.d.ts is incorrect, or this doesn't need to be `parseFloat`'d.

Reply:
Sorry, those screenshots were of durationSeconds, but delaySeconds has the same data definition.

Reply:
Even if there were user triggers returning a string instead of a number, this would still work.

![image](https://user-images.githubusercontent.com/6119598/121779720-1fc6e600-cb6b-11eb-970a-81e8f0619fb8.png)

I think it's better for the data definition to be correct in this case so that built-in triggers have enforced data types?

Reply:
I think that it's pretty unlikely that this would happen. Maybe it makes sense to check for non-number as a separate `if` case and throw a warning to the console, but I don't think it's a good idea to just blindly `parseFloat` because of other issues with string parsing, e.g. these two edge cases for locale differences and invalid strings would need to be checked:

![image](https://user-images.githubusercontent.com/6119598/121787161-91fef100-cb92-11eb-9610-ee6668867cf8.png)

Reply:
I think durations are all written locale independent these days, for what it's worth.

I was a bit torn about whether this was a hypothetical thing to support, but I think that sounds good.  We can leave this as-is and let's just accept numbers.

Reply:
I decided to do some more thorough testing, running this code through current triggers via raidemulator:
```javascript
numProps = ['delaySeconds', 'durationSeconds', 'suppressSeconds', 'soundVolume'];
testMatches = {
  castTime: '1.1',
  duration: '2.2',
};
testData = {
  // From a11s Limit Cut
  limitCutDelay: 8.8,
  // From e8s Shadow Servant Cleave Drop
  gigaSlashCleaveDebuffDuration: '1.1',
};
window.raidEmulator.popupText.triggerSets.forEach((set) => {
  if (set.triggers) {
    set.triggers.forEach((t) => {
      numProps.forEach((p) => {
        if (t[p] !== undefined) {
          let v;
          if (typeof t[p] === 'function')
            v = t[p](testData, testMatches);
          else
            v = t[p];
          if (typeof v !== 'number')
            console.log(t, p, typeof v, v);
        }
      });
    });
  }
});
```

Shows this output:
```
Object { id: "E8S Wyrmclaw", ... }
 durationSeconds string 2.2 debugger eval code:23:21
Object { id: "E8S Wyrmfang", ... }
 durationSeconds string 2.2 debugger eval code:23:21
Object { id: "E10S Giga Slash Shadow Drop Right", … }
 durationSeconds string 1.1 debugger eval code:23:21
Object { id: "E10S Giga Slash Shadow Drop Left",  … }
 durationSeconds string 1.1
```

So I think those triggers should be cleaned up first before this PR is merged?

Reply:
Forgot to mention, I checked `timelineTriggers` with the same code, adding `beforeSeconds` to the props to check, no issues with those triggers.